TactilityKernel/source/filesystem/file_lock.cpp

```
@@ -0,0 +1,67 @@
+
+    // Store a copy of the entry
+    mutex_entries.push_back({
+        .path = path,
+        .mutex = *mutex
+    });
+}
+
+void file_get_mutex(const char* path, FileMutex* mutex) {
+    for (auto& entry : mutex_entries) {
+        if (entry.path.rfind(path) == 0) {
+            memcpy(mutex, &entry.mutex, sizeof(FileMutexEntry));
```

Cppcheck (2.21.0)

[error] Using 'memcpy' on struct that contains a 'std

(memsetClass)